sunpy/map/mapbase.py

```diff
@@ -243,7 +246,51 @@ def __repr__(self):
                                refpix=u.Quantity(self.reference_pixel),
                                refcoord=u.Quantity((self.reference_coordinate.data.lon,
                                                     self.reference_coordinate.data.lat)),
-                               tmf=TIME_FORMAT) + self.data.__repr__()
+                               tmf=TIME_FORMAT)
+
+    def __repr__(self):
+        return object.__repr__(self) + "\n" + self._text_summary() + "\n" + self.data.__repr__()
+
+    def _repr_html_(self):
+        buf = BytesIO()
+        fig = Figure(figsize=(4.8, 4.8), constrained_layout=True)
+        ax = fig.subplots()
+        ax.imshow(self.data, cmap='gray', origin='lower',
+                  norm=ImageNormalize(stretch=HistEqStretch(self.data)))
+        ax.set_xlabel('X pixel')
+        ax.set_ylabel('Y pixel')
+        ax.set_title('Histogram-equalized image')
+        fig.savefig(buf, format='png')
+        image_src = b64encode(buf.getvalue()).decode('utf-8')
+
+        buf = BytesIO()
+        fig = Figure(figsize=(4.8, 2.4), constrained_layout=True)
+        ax = fig.subplots()
+        ax.hist(self.data.ravel(), bins=100, histtype='stepfilled')
+        ax.semilogy()
+        ax.xaxis.get_major_formatter().set_powerlimits((-3, 4))
```

Comment:
What's the reasoning behind manually setting this?

Reply:
It's possibly a side effect of using `constrained_layout=True`, but the tick labels were overlapping each other.  According to [the docstring for `ScalarFormatter`](https://matplotlib.org/api/ticker_api.html#matplotlib.ticker.ScalarFormatter.set_powerlimits), this setting of the power limits is an old default.

I welcome suggestions for less hacky alternatives.